packages/@ember/-internals/glimmer/lib/helpers/custom.ts

```diff
@@ -63,12 +74,94 @@ function hasDestroyable(manager: HelperManager): manager is HelperManagerWithDes
   return manager.capabilities.hasDestroyable;
 }
 
+let ARGS_CACHES = DEBUG ? new WeakMap<SimpleArgsProxy, Cache<Partial<Arguments>>>() : undefined;
+
+function getArgs(proxy: SimpleArgsProxy): Partial<Arguments> {
+  return getValue(DEBUG ? ARGS_CACHES!.get(proxy)! : proxy.argsCache!)!;
+}
+
+class SimpleArgsProxy {
+  argsCache?: Cache<Partial<Arguments>>;
+
+  constructor(
```

Comment:
Needs to be frozen, current helpers don't see this object but those using a custom manager could